scipy/signal/signaltools.py

```diff
@@ -501,17 +621,38 @@ def convolve(in1, in2, mode='full'):
         # Convolution is commutative; order doesn't have any effect on output
         volume, kernel = kernel, volume
 
+    if method == 'fft' and not _numeric_arrays([volume, kernel]):
+        raise ValueError('convolve can only be called with method=="fft" when'
+                         ' arrays consist of numeric elements (i.e., '
+                         'int/float/etc)')
+
+    if method == 'auto':
+        method = 'fft' if _fftconv_faster(volume, kernel, mode) else 'direct'
+        method = 'direct' if not _numeric_arrays([volume, kernel]) else method
+
+    if method in ('fft', 'auto'):
```

Comment:
change to

```
    if hasattr(np, "complex256") and method in ('fft', 'auto'):
```

to fix the exception on machines that don't have complex256.  though it might be better to figure out exactly which types fftconvolve supports and test for those instead.